### PR TITLE
[HDRP] Stacklit RTR: Fix issues with RTR reflections from stacklit materials

### PR DESCRIPTION
**Pull request by slunity**
---
### Purpose of this PR

Stacklit RTR: 

- Fix issues where reflections from stacklit materials were only showing lighted with indirect diffuse light and other improvements in performance mode. 
- Also fix indirect illumination lighting not tinted with stack tint for base layer.
- Also improve roughness and coatmask distillation.

---
### Testing status

Compare results with and without changes in a local test scene.

StackLit RTR performance mode no specular response
StackLit RTR performance mode no specular response 2

Fixed:

StackLit RTR performance mode fixed and improved
.

---
### Comments to reviewers

Note that other incoming PRs might also change graphic test results, might be worth it to bundle them together instead of changing screenshots for each.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Material/StackLit/StackLitRayTracing.hlsl [hunk 1]
```diff
@@ -74,12 +74,24 @@ void FitToStandardLit( BSDFData bsdfData
...
+    // Using coatnormal not necessarily better here depends on what each are vs geometric normal and the coat strength
+    // vs base strength. Could do something with that and specular albedos...
     outStandardlit.normalWS = bsdfData.normalWS;
-    outStandardlit.perceptualRoughness = bsdfData.perceptualRoughnessA;
+
+    // StandardLit expects diffuse color in baseColor:
+    outStandardlit.baseColor = bsdfData.diffuseColor;
     outStandardlit.fresnel0 = bsdfData.fresnel0;
-    outStandardlit.coatMask = bsdfData.coatMask;
+    outStandardlit.specularOcclusion = 1; // TODO
```
Comment (anisunity) [inline com.unity.render-pipelines.high-definition/Runtime/Material/StackLit/StackLitRayTracing.hlsl:89]: Why remove the custom specular occlusion input?

File: com.unity.render-pipelines.high-definition/Runtime/Material/StackLit/StackLitRayTracing.hlsl [hunk 2]
```diff
@@ -74,12 +74,24 @@ void FitToStandardLit( BSDFData bsdfData
...
+    // bool hasCoatNormal = HasFlag(surfaceData.materialFeatures, MATERIALFEATUREFLAGS_STACK_LIT_COAT)
+    //                     && HasFlag(surfaceData.materialFeatures, MATERIALFEATUREFLAGS_STACK_LIT_COAT_NORMAL_MAP);
+    // outStandardlit.normalWS = hasCoatNormal ? surfaceData.coatNormalWS : surfaceData.normalWS;
+    // Using coatnormal not necessarily better here depends on what each are vs geometric normal and the coat strength
+    // vs base strength. Could do something with that and specular albedos...
     outStandardlit.normalWS = bsdfData.normalWS;
-    outStandardlit.perceptualRoughness = bsdfData.perceptualRoughnessA;
+
+    // StandardLit expects diffuse color in baseColor:
+    outStandardlit.baseColor = bsdfData.diffuseColor;
```
Comment (anisunity) [inline com.unity.render-pipelines.high-definition/Runtime/Material/StackLit/StackLitRayTracing.hlsl:87]: There was a reason this was above the other attrbutes, we keep them in the same order than in StandardBSDFData

File: com.unity.render-pipelines.high-definition/Editor/Material/StackLit/ShaderGraph/ShaderPassDefine.template.hlsl [hunk 3]
```diff
@@ -59,7 +59,10 @@ $SpecularOcclusionConeFixupMethod.BoostAndTilt:                     #define _BEN
 #define OUTPUT_SPLIT_LIGHTING
 #endif
 
-#if !( (SHADERPASS == SHADERPASS_FORWARD) || (SHADERPASS == SHADERPASS_LIGHT_TRANSPORT))
+#if !( (SHADERPASS == SHADERPASS_FORWARD) || (SHADERPASS == SHADERPASS_LIGHT_TRANSPORT) \
```
Comment (anisunity) [inline com.unity.render-pipelines.high-definition/Editor/Material/StackLit/ShaderGraph/ShaderPassDefine.template.hlsl:62]: Cna you explain why this change?

File: com.unity.render-pipelines.high-definition/Runtime/Material/StackLit/StackLitRayTracing.hlsl (hunk 1 above)
Comment (slunity) [reply com.unity.render-pipelines.high-definition/Runtime/Material/StackLit/StackLitRayTracing.hlsl:89]: In all SO modes where it is non custom, the custom input is initialized to 0. Since the other modes are not handled/distilled for now, the simple fix is to just avoid SO in the reflections for now (otherwise, an SO of 0 will obviously block all specular indirect)

**Review by sebastienlagarde — APPROVED** [review]

**Comment by sebastienlagarde** [comment]
change look ok to me

**Review by anisunity — APPROVED** [review]
lgtm
